src/miner.cpp

```diff
@@ -86,6 +86,34 @@ class TxPriorityCompare
 
     bool operator()(const TxPriority& a, const TxPriority& b)
     {
+        // first of all, if we are comparing two certs, we must be sure they are ordered by
+        // quality if they refer to the same scid and epoch; this criterion is a consensus rule
+        // and overrides the others two
+        if (a.get<2>()->IsCertificate() && b.get<2>()->IsCertificate() )
+        {
+            // dynamic casting throws an exception upon failure
+            try {
+                const CScCertificate& aCert = dynamic_cast<const CScCertificate&>(*a.get<2>());
+                const CScCertificate& bCert = dynamic_cast<const CScCertificate&>(*b.get<2>());
+ 
+                if (aCert.GetScId() == bCert.GetScId() )
+                {
+                    LogPrint("sc", "%s():%d - qa=%d - qb=%d\n", __func__, __LINE__, aCert.quality, bCert.quality );
+
+                    if (aCert.epochNumber != bCert.epochNumber)
+                        assert("two certs for the same scid can not have different epochs" == 0);
+
+                    if (aCert.quality == bCert.quality)
+                        assert("two certs for the same scid can not have the same quality" == 0);
+
+                    return aCert.quality > bCert.quality;
+                }
+            } catch (...) {
+                LogPrintf("%s():%d - ERROR: cast error\n", __func__, __LINE__ );
+                assert("could not cast txbase obj" == 0);
+            }
+        }
+
```

Comment:
While appreciate the succinctness of this change, I think that
1- probably we should not assert into this `operator()`, but write an ad hoc function like 'VerifyCertificateDependencies', similarly to what we have done with `VerifyForwardTransfersDependencies` in this same file
2- 1- we should not assert right away; compare `VerifyForwardTransfersDependencies` where, in some cases we assert only if fDebug flag is set
@i-Alex @alsala what do you think about it?

Reply:
Also I'd take advantage to get rid of all the `#if 0` blocks left over from beta :-)

Reply:
As agreed offline, I will move checks to a `VerifyCertificateDependencies` and make sure to mimic assert policy used already in  `VerifyForwardTransfersDependencies `

Reply:
Done